python_bindings/bindings.cpp

```diff
@@ -254,6 +275,239 @@ class Index {
+      });
+
+      return py::make_tuple(appr_alg->offsetLevel0_,
+                            appr_alg->max_elements_,
+                            appr_alg->cur_element_count,
+                            appr_alg->size_data_per_element_,
+                            appr_alg->label_offset_,
+                            appr_alg->offsetData_,
+                            appr_alg->maxlevel_,
+                            appr_alg->enterpoint_node_,
+                            appr_alg->maxM_,
+                            appr_alg->maxM0_,
+                            appr_alg->M_,
+                            appr_alg->mult_,
+                            appr_alg->ef_construction_,
+                            appr_alg->ef_,
+                            appr_alg->has_deletions_,
+                            appr_alg->size_links_per_element_,
+                            py::array_t<hnswlib::labeltype>(
+                                    {appr_alg->label_lookup_.size()}, // shape
+                                    {sizeof(hnswlib::labeltype)}, // C-style contiguous strides for double
+                                    label_lookup_key_npy, // the data pointer
+                                    free_when_done_lb),
+                            py::array_t<hnswlib::tableint>(
+                                    {appr_alg->label_lookup_.size()}, // shape
+                                    {sizeof(hnswlib::tableint)}, // C-style contiguous strides for double
+                                    label_lookup_val_npy, // the data pointer
+                                    free_when_done_id),
+                            py::array_t<int>(
+                                    {appr_alg->element_levels_.size()}, // shape
+                                    {sizeof(int)}, // C-style contiguous strides for double
+                                    element_levels_npy, // the data pointer
+                                    free_when_done_lvl),
+                            py::array_t<char>(
+                                    {level0_npy_size}, // shape
+                                    {sizeof(char)}, // C-style contiguous strides for double
+                                    data_level0_npy, // the data pointer
+                                    free_when_done_l0),
+                            py::array_t<char>(
+                                    {link_npy_size}, // shape
+                                    {sizeof(char)}, // C-style contiguous strides for double
+                                    link_list_npy, // the data pointer
+                                    free_when_done_ll)
+                          );
+
+    }
+
+
+    py::tuple getIndexParams() const {
+        /*  TODO: serialize state of random generators appr_alg->level_generator_ and appr_alg->update_probability_generator_  */
+        /*        for full reproducibility / to avoid re-initializing generators inside Index<float>::createFromParams         */
```

Comment:
Added TODO block to serialize state of random generators. Currently, the random generators are re-initialized on index deserialization. This could make index operations not (fully) reproducible.